### PR TITLE
Factorize gcd

### PR DESCRIPTION
**Pull request by zasdfgbnm**
```c++
gcd(a * x, b * x) -> gcd(a, b) * abs(x)
```
especially if `x >= 0`:
```
gcd(a * x, x) -> x
```

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: test/test_expr_simplifier.cpp [hunk 1]
```diff
@@ -426,13 +426,18 @@ Bool* operator""_b(const char* str, size_t) {
...
+class ExprSimplifierTest : public NVFuserTest {
+  std::unique_ptr<Fusion> fusion_ptr;
+  std::unique_ptr<FusionGuard> fusion_guard_ptr;
+
+  void SetUp() override {
+    NVFuserTest::SetUp();
+    fusion_ptr = std::make_unique<Fusion>();
+    fusion_guard_ptr = std::make_unique<FusionGuard>(fusion_ptr.get());
+  }
+};
```
Comment (zasdfgbnm) [inline test/test_expr_simplifier.cpp:438]: Just doing some cleanup

File: test/test_expr_simplifier.cpp [hunk 2]
```diff
@@ -1086,14 +1029,20 @@ TEST_F(ExprSimplifierTest, Assume_CUDA) {
...
 }
 
+TEST_F(ExprSimplifierTest, FactorizeGcd_CUDA) {
+  EXPECT_TRUE(simplifyExpr("gcd( i1 * i2 , i3 * i2 )"_)
+                  ->sameAs("gcd( i1 , i3 ) * abs( i2 )"_));
+  EXPECT_TRUE(simplifyExpr("gcd( i1 * i2 , i3 * i2 )"_, {}, {"i2 >= 0"_b})
+                  ->sameAs("gcd( i1 , i3 ) * i2"_));
+  EXPECT_TRUE(simplifyExpr("gcd( i1 * i2 , i2 )"_)->sameAs("abs( i2 )"_));
+  EXPECT_TRUE(
+      simplifyExpr("gcd( i1 * i2 , i2 )"_, {}, {"i2 >= 0"_b})->sameAs("i2"_));
```
Comment (zasdfgbnm) [inline test/test_expr_simplifier.cpp:1045]: I expect this to be the most useful case

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**